### PR TITLE
Expand test coverage from 29% to 82%

### PR DESCRIPTION
**Pull request by hughsie**
Add 24 new C unit tests covering passim-common config helpers, PassimItem properties/flags/serialization, GnuTLS key and certificate generation, and PassimClient defaults. Add a Python integration test using dbusmock to mock Avahi and exercise the real passimd daemon end-to-end via HTTPS requests and D-Bus calls.

## Summary by CodeRabbit

- **New Features**
  - Added a `--insecure` startup option to relax D-Bus UID authorization checks.
  - Added an `installed_tests` Meson option (default on) to install distributed test artefacts, shipped as a new RPM `tests` subpackage.
- **Bug Fixes**
  - Improved handling and validation for edge cases like invalid/zero ports and empty SHA256 input.
- **Testing**
  - Expanded self-test coverage and added an HTTP/HTTPS + D-Bus integration test using mocked system services.
  - Added an optional client/CLI test helper for broader verification.
- **CI**
  - Updated the Fedora CI image to include Python D-Bus and D-Bus mock support.

### CODE REVIEW

File: src/passim-self-test.c [hunk 1]
```diff
@@ -135,6 +112,636 @@ passim_common_func(void)
...
+
+static void
+passim_gnutls_datum_to_dn_str_func(void)
+{
+	g_autofree gchar *str = NULL;
+	gnutls_datum_t raw = {0};
+
+	/* empty datum should return NULL (import fails) */
+	str = passim_gnutls_datum_to_dn_str(&raw);
+	g_assert_null(str);
```
[the remark this answers is not in the document]
Comment (hughsie) [reply src/passim-self-test.c:736]: The current code already handles the {.data = NULL, .size = 0} datum correctly — gnutls_x509_dn_import(dn, raw) returns a negative error code for invalid input, and line 121-122 returns NULL in that case. The test at line 718-719 confirms this works. Adding a redundant NULL/size check before the dn_import call would be defensive code with no behavioral change.

File: src/passim-integration-test.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+    @classmethod
+    def _install_dbus_interface(cls, srcdir):
+        dbus_iface_dir = os.path.join(cls.datadir, "dbus-1", "interfaces")
+        os.makedirs(dbus_iface_dir, exist_ok=True)
+        dbus_iface_dst = os.path.join(
+            dbus_iface_dir, "org.freedesktop.Passim.xml"
+        )
+        dbus_iface_src = os.path.join(srcdir, "org.freedesktop.Passim.xml")
+        if not os.path.exists(dbus_iface_dst):
+            shutil.copy2(dbus_iface_src, dbus_iface_dst)
```
[the remark this answers is not in the document]
Comment (hughsie) [reply src/passim-integration-test.py:180]: The conditional on line 132 means we only copy when the file doesn't already exist — i.e. meson install already put it there. Deleting it on cleanup would
  remove the legitimately installed file. The copy is a no-op in practice (the build installs it to the same prefix), and when it does copy, the file belongs in that
  location permanently. Registering cleanup to delete it would be wrong.
